src/MultiselectAccessibility.vue

```diff
@@ -0,0 +1,141 @@
+    ${multiple ? ', press left to focus selected values' : ''}
+  `
+}
+
+export default {
+  props: {
+    options: {
+      type: Array,
+      required: true
+    },
+    pointer: {
+      type: Number,
+      required: true
+    },
+    value: {
+      type: null,
+      default: () => []
+    },
+    isFocused: {
+      type: Boolean,
+      default: false
+    },
+    isOpen: {
+      type: Boolean,
+      required: true
+    },
+    searchable: {
+      type: Boolean,
+      default: false
+    },
+    multiple: {
+      type: Boolean,
+      default: false
+    },
+    a11yMessages: {
+      type: Function,
+      default: () => ({})
+    },
+    getOptionLabel: {
+      type: Function,
+      default: () => ''
+    }
+  },
+  data () {
+    return {
+      tmpAnnouncement: []
+    }
+  },
+  computed: {
+    currentlyFocusedElement () {
+      return this.options[this.pointer]
+    },
+    currentlyFocusedElementLabel () {
+      return this.getOptionLabel(this.currentlyFocusedElement)
+    },
+    selectedElementAnnouncement () {
+      return `Option ${this.currentlyFocusedElementLabel} is focused`
+    },
+    currentValueAnnouncement () {
+      const valueLabels = this.value.map(this.getOptionLabel)
+      return `Selected options: ${valueLabels.join(', ')}`
+    },
+    currentAnnouncement () {
+      const valueLabels = this.value.map(this.getOptionLabel)
+
+      const messages = {
+        ...DEFAULT_MESSAGES,
+        ...this.a11yMessages
+      }
+      const status = this.isOpen
+        ? 'input'
+        : 'menu'
+
+      return messages[status]({
+        searchable: this.searchable,
+        multiple: this.multiple,
+        disabled: this.disabled,
+        options: this.options,
+        pointer: this.pointer,
+        focusedElement: this.currentlyFocusedElementLabel,
+        valueLabels
+      })
+    }
+  },
+  watch: {
+    value (newValue, oldValue) {
+      const newValueLabels = newValue.map(this.getOptionLabel)
+      const oldValueLabels = oldValue.map(this.getOptionLabel)
+      let toRemove = 0
+
+      if (newValueLabels.length > oldValueLabels.length) {
+        const addedValues = newValueLabels.filter(label => !oldValueLabels.includes(label)).map(label => `${label} added`)
+        this.tmpAnnouncement.push(...addedValues)
+        toRemove = addedValues.length
+
+      } else if (newValueLabels.length < oldValueLabels.length) {
+        const removedValues = oldValueLabels.filter(label => !newValueLabels.includes(label)).map(label => `${label} removed`)
+        this.tmpAnnouncement.push(...removedValues)
+        toRemove = removedValues.length
+      } else if (newValueLabels.length === 1 && oldValueLabels.length === 1){
+        this.tmpAnnouncement.push(...newValueLabels).map(label => `${label} selected`)
+        toRemove = 1
+      }
+
+      setTimeout(() => {
+        this.tmpAnnouncement.splice(0, toRemove)
+      }, 1500);
+    }
+  }
+}
+</script>
+
+<style>
+.multiselect__visually-hidden {
```

Comment:
Here's a better, more widely accepted css method for sr-only:

`.sr-only:not(:focus):not(:active) {
  clip: rect(0 0 0 0); 
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap; 
  width: 1px;
}`